src/Core/Checkout/Cart/Event/LineItemAddedFinishEvent.php

```diff
@@ -0,0 +1,60 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Cart\Event;
+
+use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Framework\Event\ShopwareSalesChannelEvent;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+class LineItemAddedFinishEvent implements ShopwareSalesChannelEvent
+{
+    /**
+     * @var array
+     */
+    protected $lineItems;
+
+    /**
+     * @var Cart
+     */
+    protected $cart;
+
+    /**
+     * @var SalesChannelContext
+     */
+    protected $context;
+
+    /**
+     * @var bool
+     */
+    protected $merged;
+
+    public function __construct(array $lineItems, Cart $cart, SalesChannelContext $context)
+    {
+        $this->lineItems = $lineItems;
+        $this->cart = $cart;
+        $this->context = $context;
+    }
+
+    public function getLineItems(): array
+    {
+        return $this->lineItems;
+    }
+
+    public function getCart(): Cart
+    {
+        return $this->cart;
+    }
+
+    /**
+     * @deprecated tag:v6.4.0 - Will return Shopware\Core\Framework\Context instead
+     */
+    public function getContext(): SalesChannelContext
```

Comment:
this return type is against the definition from the interface. this should break on your machine right now :blush: 

please also remove the deprecation message